infinispan/src/main/java/com/yahoo/ycsb/db/RemoteCacheManagerHolder.java

```diff
@@ -0,0 +1,21 @@
+package com.yahoo.ycsb.db;
+
+import java.util.Properties;
+
+import org.infinispan.client.hotrod.RemoteCacheManager;
+
+public class RemoteCacheManagerHolder {
+	
+	private static RemoteCacheManager cacheManager = null;
+	
+	private RemoteCacheManagerHolder() {}
+	
+	public static RemoteCacheManager getInstance(Properties props){
```

Comment:
this doesn't ensure only a single RemoteCacheManager is made and it unsafely publishes cacheManager before it is initialized.

either synchronize the getInstance method or use a proper double-check idiom and make cacheManager volatile.